src/Transition.js

```diff
@@ -238,38 +244,34 @@ class Transition extends React.Component {
     this.props.onEnter(node, appearing);
 
     this.safeSetState({ status: ENTERING }, () => {
-      this.props.onEntering(node, appearing);
+      this.props.onEntering(node, appearing, setEntered);
```

Comment:
won't e.g. `onEntered` get called twice if the user both manually triggers the transition end _and_ the timeout fires? i think, then, that this should hook into that weird "next callback" mechanism. i wish i clearly remembered how that actually worked well enough to give you a real suggestion on implementation, though